src/main/java/org/mifos/pheevouchermanagementsystem/service/VoucherLifecycleService.java

```diff
@@ -0,0 +1,168 @@
+    {
+        voucherInstructionList.stream().forEach(voucherInstruction -> {
+            String requestID = request.getRequestID();
+            Boolean voucherValidation = validateVoucherForSuspend(voucherInstruction, request);
+            if (voucherValidation) {
+                Voucher voucher = voucherRepository.findBySerialNo(voucherInstruction.getSerialNumber()).orElseThrow(() -> VoucherNotFoundException.voucherNotFound(voucherInstruction.getSerialNumber()));
+                if(validateRegisteringInstitutionId(voucher, registeringInstitutionId)) {
+                    changeStatus(voucher, SUSPENDED.getValue());                }
+                else {
+                    FailedCaseDTO failedCase =   new FailedCaseDTO(voucherInstruction.getSerialNumber(), "The voucher provided in the request does not correspond to the same registering ID.");
+                    failedCaseList.add(failedCase);
+                }
+            } else {
+                FailedCaseDTO failedCase =   new FailedCaseDTO(voucherInstruction.getSerialNumber(), "Voucher validation failed.");
+                failedCaseList.add(failedCase);
+            }
+        });
+    }
+
+    public void validateAndCancelVouchers(List<VoucherInstruction> voucherInstructionList, RequestDTO request, List<FailedCaseDTO> failedCaseList, String registeringInstitutionId){
+        voucherInstructionList.stream().forEach(voucherInstruction -> {
+            String requestID = request.getRequestID();
+            Boolean voucherValidation = validateVoucherForCancel(voucherInstruction, request);
+            if (voucherValidation) {
+                Voucher voucher = voucherRepository.findBySerialNo(voucherInstruction.getSerialNumber()).orElseThrow(() -> VoucherNotFoundException.voucherNotFound(voucherInstruction.getSerialNumber()));
+                if(validateRegisteringInstitutionId(voucher, registeringInstitutionId)) {
+                    changeStatus(voucher, CANCELLED.getValue());                }
+                else {
+                    FailedCaseDTO failedCase =   new FailedCaseDTO(voucherInstruction.getSerialNumber(), "The voucher provided in the request does not correspond to the same registering ID.");
+                    failedCaseList.add(failedCase);
+                }
+            } else {
+                FailedCaseDTO failedCase =   new FailedCaseDTO(voucherInstruction.getSerialNumber(), "Voucher validation failed.");
+                failedCaseList.add(failedCase);
+            }
+        });
+    }
+
+    public Boolean validateVoucherForReactivate(VoucherInstruction voucherInstruction, RequestDTO request){
+        if(voucherRepository.existsBySerialNo(voucherInstruction.getSerialNumber())){
+            Voucher voucher = voucherRepository.findBySerialNo(voucherInstruction.getSerialNumber()).orElseThrow(() -> VoucherNotFoundException.voucherNotFound(voucherInstruction.getSerialNumber()));
```

Comment:
in every method we have a lot of status checks, which is difficult to manage
check if you can write a simple state machine to validate these 
ref: https://www.baeldung.com/java-enum-simple-state-machine

Reply:
this will require some effort can we add this task to backlog